src/graphql/lib/resolvers/getByDOI.js

```diff
@@ -101,7 +99,7 @@ const getByDOI = {
       try {
         await redisClient.set(apikey, `${JSON.stringify(apiKeyConfig)}`);
       } catch (err) {
-        logger.error(`Cannot update apikey [${apikey}] for [count]`);
+        logger.error(`[redis] Cannot update apikey [${apikey}] with config [${JSON.stringify(apiKeyConfig)}]`, err);
         return Promise.reject(err);
```

Comment:
You're in an async function, just `throw` error
```suggestion
    throw err;
```

(Sorry for duplicate, but it's just this one in this file)